networkx/algorithms/centrality/domirank.py

```diff
@@ -0,0 +1,342 @@
+        time cost of the 'analytical' method is large for
+        non-regular graphs, but provides the true DomiRank.
+
+    alpha: float, optional (default=0.95)
+        The level of competition for DomiRank.
+
+    dt: float, optional (default=0.1)
+        The step size for the Newton iteration.
+
+    epsilon: float, optional (default=1e-5)
+        The relative stopping criterion for convergence.
+
+    max_iter: integer, optional (default=100)
+        Maximum number of Newton iterations allowed.
+        It is recommended that ''max_iter >= 50''.
+
+    patience: integer, optional (default=10)
+        The number of steps between convergence checks.
+        It is recommended that ''patience >= 10''.
+
+
+    Returns
+    -------
+    nodes : dictionary
+        Dictionary keyed by node with DomiRank centrality of the node as value.
+
+    sigma : float
+        $\alpha$ normalized by the smallest eigenvalue.
+
+    converged : boolean
+        Whether the centrality computation converged. Returns ``None`` if ``comp_method = `analytical```.
+
+    Examples
+    --------
+    >>> G = nx.path_graph(5)
+    >>> centrality, sigma, converged = nx.domirank(G)
+    >>> print([f"{node} {centrality[node]:0.2f}" for node in centrality])
+    ['0 -0.54', '1 1.98', '2 -1.08', '3 1.98', '4 -0.54']
+
+    Raises
+    ------
+    NetworkXPointlessConcept
+        If the graph `G` is the null graph.
+
+    NetworkXUnfeasible
+        If alpha is negative (and thus outside its bounds): ``alpha < 0``.
+
+        If ``alpha > 1`` with the ``comp_method = `iterative``` argument.
+
+    NetworkXAlgorithmError
+        If the comp_method is not of type: {'analytical', 'iterative'}.
+
+        If ``patience > max_iter``.
+
+        If ``max_iter < 5``.
+
+        If dt is not in the bounds: ``0 < dt < 1``.
+
+        If epsilon is negative or equal to one: ``epsilon <= 0``.
+
+    Warning
+        If supercharging the competition parameter for the analytical solution: ``alpha > 1``.
+
+        If one is using the analytical solution (comp_method=`analytical`) for a large system, i.e. ``len(G) > 5000``, as the algorithm will be slow.
+
+    See Also
+    --------
+    :func:`~networkx.algorithms.centrality.degree_centrality`
+    :func:`~networkx.algorithms.link_analysis.pagerank_alg.pagerank`
+    :func:`~networkx.algorithms.link_analysis.hits_alg.hits`
+
+    References
+    ----------
+    .. [1] Engsig, M., Tejedor, A., Moreno, Y. et al.
+        "DomiRank Centrality reveals structural fragility of complex networks via node dominance."
+        Nat Commun 15, 56 (2024). https://doi.org/10.1038/s41467-023-44257-0
+    """
+    import numpy as np
+    import scipy as sp
+
+    if len(G) == 0:
+        raise nx.NetworkXPointlessConcept(
+            "cannot compute centrality for the null graph."
+        )
+    if patience > max_iter:
+        raise nx.NetworkXAlgorithmError("it is mandatory that `max_iter > patience`.")
```

Comment:
Another general comment for the error messages: avoid using backticks, as they won't render properly anyways and just end up polluting the traceback.